From 42e77cbbe9a52b2ef0b15097ff1fdcd951c9b8ee Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Fri, 16 Oct 2026 22:39:40 +0800
Subject: [PATCH 49/49] loonggpu-bridge: add missing sentinel in
 special_displays[]

Dereferencing the pointer to "the element after the last element of an
array" invokes undefined behavior, so it "worked" with GCC 15 but now
fails with GCC 16.

Man, what can I say.

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu-bridge/bridge_phy.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/loonggpu-bridge/bridge_phy.c b/loonggpu-bridge/bridge_phy.c
index 02cb830..16ef552 100644
--- a/loonggpu-bridge/bridge_phy.c
+++ b/loonggpu-bridge/bridge_phy.c
@@ -126,6 +126,7 @@ static const struct special_display special_displays[] = {
 		.serial = 16843009,
 		.sd_func = fix_vsc_missing_display,
 	},
+	{ /* sentinel */ },
 };
 
 void is_special_display(struct loonggpu_device *adev, struct loonggpu_connector *aconnector, u32 link)
-- 
2.56.0

